packages/@glimmer/compiler/lib/compiler.ts

```diff
@@ -116,10 +112,20 @@ export function precompile(
     id: idFn(JSON.stringify(options.meta) + blockJSON),
     block: blockJSON,
     moduleName: moduleName ?? '(unknown template module)',
+    scope: null,
+    isStrictMode: options.strictMode ?? false,
   };
 
   // JSON is javascript
-  return JSON.stringify(templateJSONObject);
+  let stringified = JSON.stringify(templateJSONObject);
+
+  if (usedLocals.length > 0) {
+    let scopeFn = `()=>[${usedLocals.join(',')}]`;
+
+    stringified = stringified.replace('"scope":null', `"scope":${scopeFn}`);
```

Comment:
This seems somewhat brittle to me. Is there a different / better way? One would be to add `scope: 'SOME-GUID-VALUE'` and then we know precisely what to replace...

Reply:
yeah, I can do that, sounds good 👍